packages/btcindexer/src/btcindexer.ts

```diff
@@ -455,6 +441,39 @@ export class Indexer {
 							status: MintTxStatus.MintFailed,
 						})),
 					);
+				} else {
+					const [success, suiTxDigest] = result;
+
+					if (success) {
+						logger.info({
+							msg: "Sui batch mint transaction successful",
+							suiTxDigest,
+							pkgKey,
+						});
+						await this.storage.batchUpdateNbtcTxs(
+							processedPrimaryKeys.map((p) => ({
+								txId: p.tx_id,
+								vout: p.vout,
+								status: MintTxStatus.Minted,
+								suiTxDigest,
+							})),
+						);
+					} else {
+						logger.error({
+							msg: "Sui batch mint transaction failed (on-chain execution error)",
+							method: "processFinalizedTransactions",
+							suiTxDigest,
+							pkgKey,
+						});
+						await this.storage.batchUpdateNbtcTxs(
+							processedPrimaryKeys.map((p) => ({
+								txId: p.tx_id,
+								vout: p.vout,
```

Comment:
processedPrimaryKeys will have only one tx_id (and possible many vouts) or many txs?

maybe we should consider changing the column type for vout to be a list (not in this tasks, but let's think about it).